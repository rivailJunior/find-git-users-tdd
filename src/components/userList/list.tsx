import { iUser } from "../../interfaceModel/user";
import TableList from '../table'

type iList = {
    loading: boolean;
    users: iUser[]
}

export default function List({ loading, users }: iList) {
    return (
        <div>
            <h3>Lista de Usu√°rios</h3>
            {loading ? (
                <div>Loading...</div>
            ) : (

                <TableList
                    data={users}
                    headerProperties={[{ label: 'Login' }]}
                    bodyProperties={[{ key: 'login' }]}
                />
            )}
        </div>
    );
}
