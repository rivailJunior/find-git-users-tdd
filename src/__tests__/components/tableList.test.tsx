import { render, screen } from '@testing-library/react'
import TableList from '../../components/table'
import { userList } from '../../mock/userList'

const Wrapper = () => {
    return (
        <TableList>
            <TableList.Header headerProperties={[{ label: 'Login' }]} />
            <TableList.Body bodyProperties={[{ key: 'login' }]} data={userList} />
        </TableList>
    )
}

describe('Render Table', () => {
    it('Snapshot', () => {

        const { container } = render(<Wrapper />);
        expect(container.firstChild).toMatchSnapshot()
    });

    // describe('When try pass a different size of objects to header and body', () => {
    //     it('Show an diff size error', () => {
    //         const header = [
    //             { label: 'Login' },
    //             { label: 'Email' },
    //         ];
    //         expect(() => render(<TableList bodyProperties={[{ key: 'login' }]} data={userList} headerProperties={header} />)).toThrowError('Size of header columns is different of body columns.');
    //     });
    // })
})