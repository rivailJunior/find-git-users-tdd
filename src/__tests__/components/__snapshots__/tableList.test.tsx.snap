// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Table Snapshot 1`] = `
<table
  class="table table-striped table-bordered table-hover"
>
  <thead
    data-testid="tableHeader"
  >
    <tr>
      <th>
        Login
      </th>
    </tr>
  </thead>
  <tbody
    data-testid="userList"
  >
    <tr>
      <td>
        mojombo
      </td>
    </tr>
    <tr>
      <td>
        defunkt
      </td>
    </tr>
    <tr>
      <td>
        pjhyett
      </td>
    </tr>
  </tbody>
</table>
`;
